Telerik.JustMock/Core/Context/MSpecContextResolver.cs

```diff
@@ -36,49 +36,52 @@ public MSpecContextResolver()
 
 		public override MocksRepository ResolveRepository(UnresolvedContextBehavior unresolvedContextBehavior)
 		{
-			lock (this.repositorySync)
-			{
-				var stackTrace = new StackTrace();
-				var frames = stackTrace.EnumerateFrames().ToList();
-				var testMethod = FindExistingTestMethod(frames);
-				if (testMethod != null)
-					return repositories[testMethod.DeclaringType];
-				if (unresolvedContextBehavior == UnresolvedContextBehavior.DoNotCreateNew)
-					return null;
-
-				var caller = frames.FirstOrDefault(method => method.Module.Assembly != typeof(MocksRepository).Assembly);
-				var mspecTestClass = caller.DeclaringType;
-
-				MocksRepository parentRepo;
-				repositories.TryGetValue(mspecTestClass.BaseType, out parentRepo);
-
-				var repo = new MocksRepository(parentRepo, caller);
-				repositories.Add(mspecTestClass, repo);
-
-				return repo;
-			}
+            var testMethod = this.GetTestMethod();
```

Comment:
@lyubomir-rusev Please put back synchronized section, it a part of another feature and has to be presented.